source/index.ts

```diff
@@ -57,7 +57,7 @@ export const enum TypeName {
 
 const toString = Object.prototype.toString;
 const isOfType = <T>(type: string) => (value: unknown): value is T => typeof value === type;
-const isBuffer = (input: any): input is Buffer => !is.nullOrUndefined(input) && !is.nullOrUndefined(input.constructor) && is.function_(input.constructor.isBuffer) && input.constructor.isBuffer(input);
+const isBuffer = (input: unknown): input is Buffer => !is.nullOrUndefined(input) && !is.nullOrUndefined((input as Buffer).constructor) && is.function_((input as Buffer).constructor.isBuffer) && (input as Buffer).constructor.isBuffer(input);
```

Comment:
Why do we need the `as Buffer` when we already have the `input is Buffer` type guard?

Reply:
Well the typeguard `is Buffer` is more like a return type, so the code placed *after* the invocation `.isBuffer(input)` will know if `input` is a `Buffer` or not but within the body this is just being evaluted. Also we're using custom typeguards thus sort of overriding the default Typescript behavior (helping by explicitly stating what it can't find out on its own 🙏)

Reply:
Good point 👍